nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py

```diff
@@ -149,325 +162,295 @@ def create_site_location_type_locations(
-            jh.content_types.add(location_ct)
-
-        Note = apps.get_model("extras", "note")
-        notes = Note.objects.filter(assigned_object_type=region_ct)
-        for note in notes:
-            note.assigned_object_type = location_ct
-            region = Region.objects.get(id=note.assigned_object_id)
-            region_loc = Location.objects.get(name=region.name, location_type=region_lt)
-            note.assigned_object_id = region_loc.id
-        Note.objects.bulk_update(notes, ["assigned_object_type", "assigned_object_id"], 1000)
-
-        Relationship = apps.get_model("extras", "relationship")
-        RelationshipAssociation = apps.get_model("extras", "relationshipassociation")
-
-        src_relationship_associations = RelationshipAssociation.objects.filter(relationship__source_type=region_ct)
-        src_relationship_associations.update(source_type=location_ct)
-        for relationship_association in src_relationship_associations:
-            src_region = Region.objects.get(id=relationship_association.source_id)
-            src_loc = Location.objects.get(name=src_region.name, location_type=region_lt)
-            relationship_association.source = src_loc
-            relationship_association.source_id = src_loc.id
-        RelationshipAssociation.objects.bulk_update(src_relationship_associations, ["source_id"], 1000)
-        src_relationships = Relationship.objects.filter(source_type=region_ct)
-        src_relationships.update(source_type=location_ct)
-
-        dst_relationship_associations = RelationshipAssociation.objects.filter(relationship__destination_type=region_ct)
-        dst_relationship_associations.update(destination_type=location_ct)
-        for relationship_association in dst_relationship_associations:
-            dst_region = Region.objects.get(id=relationship_association.destination_id)
-            dst_loc = Location.objects.get(name=dst_region.name, location_type=region_lt)
-            relationship_association.destination = dst_loc
-            relationship_association.destination_id = dst_loc.pk
-        RelationshipAssociation.objects.bulk_update(dst_relationship_associations, ["destination_id"], 1000)
-        dst_relationships = Relationship.objects.filter(destination_type=region_ct)
-        dst_relationships.update(destination_type=location_ct)
-
-        WebHook = apps.get_model("extras", "webhook")
-        web_hooks = WebHook.objects.filter(content_types__in=[region_ct])
-        for wh in web_hooks:
-            wh.content_types.add(location_ct)
-
-    # Reassign Site Models to Locations of Site LocationType
-    if Site.objects.exists():  # Iff Site instances exist
+    model_class = apps.get_model("dcim", f"{model}")
+    model_ct = ContentType.objects.get_for_model(model_class)
+    if model == "region":
+        model_lt = LocationType.objects.get(name="Region")
+    else:
+        model_lt = LocationType.objects.get(name="Site")
+
+        # Only Site Related models
```

Comment:
Another non-blocking nitpick: maybe move all of this into a `reassign_site_model_instances_to_locations`